doc/api/quic.md

```diff
@@ -0,0 +1,1375 @@
+
+#### `endpoint.closed`
+
+* Type: {Promise} A promise that is resolved when the `Endpoint` has been
+  closed, or is rejected if the `Endpoint` is abnormally terminated. This
+  will be the same promise that is returned by the `endpoint.close()`
+  function.
+
+#### `endpoint.closing`
+
+* Type: {boolean} Set to `true` after `endpoint.close()` has been
+  called and the `Endpoint` has not yet been destroyed.
+
+#### `endpoint.connect(address[, options[, resume]])`
+
+* `address` {net.SocketAddressInit|net.SocketAddress}
+* `options` {quic.SessionConfigInit|quic.SessionConfig}
+* `resume` {Object} A saved TLS session ticket as provided by the
+  `session.sessionTicket` property of a previously established {quic.Session}.
+  * `sessionTicket` {ArrayBuffer} The serialized TLS session ticket.
+  * `transportParams` {ArrayBuffer} The serialized remote transport parameters.
+* Returns: {quic.Session}
+
+Creates a new client {quic.Session} and begins the TLS handshake
+to the identified remote peer.
+
+The `address` argument is either a {net.SocketAddress} or a
+{net.SocketAddressInit} object used to create a new {net.SocketAddress}.
+
+If the `resume` argument is provided, the `Endpoint` will attempt to use the
+provided TLS session ticket or transport parameters to resume a previously
+established TLS session. If the provided information cannot be used, or the
+session cannot be resumed, the provided information will be ignored.
+
+#### `endpoint.destroy([error])`
+
+* `error` {Error}
+
+Immediately and synchronously destroys the `Endpoint`. Once destroyed,
+the `endpoint.closed` promise will be fulfilled and the `Endpoint` will
+no longer be usable.
+
+#### `endpoint.listen(options)`
+
+* `options` {quic.SessionConfigInit|quic.SessionConfig}
+
+Causes the `Endpoint` to begin listening to inbound initial QUIC packets.
+
+#### `endpoint.listening`
+
+* Type: {boolean} Set to `true` after `endpoint.listen()` has been
+  called successfully and before the `Endpoint` has been destroyed.
+
+#### `endpoint.stats`
+
+* Type: {quic.EndpointStats}
+
+Returns an object providing access to statistics accumulated for the
+`Endpoint`. After the `Endpoint` has been destroyed, the stats provide
+the most details available at the time it was destroyed.
+
+#### Cloning `Endpoint` using `MessagePort` or `BroadcastChannel`
+
+The `Endpoint` object is cloneable using the Structured Clone Algorithm.
+However, there are a number of details about how the object is cloned that
+are important.
+
+When an `Endpoint` is first created, the underlying UDP socket handle is created
+and bound to the context and event loop associated with the current thread. A
+cloned `Endpoint` may exist in a separate thread with a separate event loop but
+will still use the original UDP socket handle in the original event loop to
+transmit or receive packets.
+
+<!-- lint disable nodejs-links -->
+![Cloned QUIC Endpoints](assets/cloned-quic-endpoint.png)
+<!-- lint enable nodejs-links -->
```

Comment:
The linter error would to resolved if https://github.com/nodejs/node/pull/39206 lands.